tests/bgp/conftest.py

```diff
@@ -24,18 +27,49 @@ def setup_bgp_graceful_restart(duthost, nbrhosts):
-        logger.info("enable graceful restart on neighbor {}".format(k))
-        logger.info("bgp asn {}".format(nbr['conf']['bgp']['asn']))
-        nbr['host'].eos_config(lines=["graceful-restart restart-time 300"], \
-                               parents=["router bgp {}".format(nbr['conf']['bgp']['asn'])])
-        nbr['host'].eos_config(lines=["graceful-restart"], \
-                               parents=["router bgp {}".format(nbr['conf']['bgp']['asn']), "address-family ipv4"])
-        nbr['host'].eos_config(lines=["graceful-restart"], \
-                               parents=["router bgp {}".format(nbr['conf']['bgp']['asn']), "address-family ipv6"])
-
-    # change graceful restart option will clear the bgp session.
-    # so, let's wait for all bgp sessions to be up
+    def configure_nbr_gr(node=None, results=None):
+        """Target function will be used by multiprocessing for configuring VM hosts.
+
+        Args:
+            node (object, optional): A value item of the dict type fixture 'nbrhosts'. Defaults to None.
+            results (Proxy to shared dict, optional): An instance of multiprocessing.Manager().dict(). Proxy to a dict
+                shared by all processes for returning execution results. Defaults to None.
+        """
+        if node is None or results is None:
+            logger.error('Missing kwarg "node" or "results"')
+            return
+
+        node_results = []
+        logger.info('enable graceful restart on neighbor host {}'.format(node['host'].hostname))
+        logger.info('bgp asn {}'.format(node['conf']['bgp']['asn']))
+        node_results.append(node['host'].eos_config(
+                lines=['graceful-restart restart-time 300'], \
+                parents=['router bgp {}'.format(node['conf']['bgp']['asn'])], \
+                module_ignore_errors=True)
+            )
+        node_results.append(node['host'].eos_config(
+                lines=['graceful-restart'], \
+                parents=['router bgp {}'.format(node['conf']['bgp']['asn']), 'address-family ipv4'], \
+                module_ignore_errors=True)
+            )
+        node_results.append(node['host'].eos_config(
+                lines=['graceful-restart'], \
+                parents=['router bgp {}'.format(node['conf']['bgp']['asn']), 'address-family ipv6'], \
+                module_ignore_errors=True)
+            )
+        results[node['host'].hostname] = node_results
+
+    results = parallel_run(configure_nbr_gr, (), {}, nbrhosts.values(), timeout=120)
+
+    failed_results = {}
+    for node_name, node_results in results.items():
+        failed_node_results = [res for res in node_results if res['failed']]
+        if len(failed_node_results) > 0:
+            failed_results[node_name] = failed_node_results
+    if failed_results:
+        logger.error('failed_results => {}'.format(json.dumps(failed_results, indent=2)))
+        pt_assert(False, 'Some processes for configuring nbr hosts returned failed results')
```

Comment:
line 64-71 is most duplicated in two places, wonder if you can provide helper functions to ease the result check.

Reply:
Sure, updated.